File: gla-ops-application/src/main/ui/app/scripts/services/AuditService.js
/**
 * Copyright (c) Greater London Authority, 2016.
 *
 * This source code is licensed under the Open Government Licence 3.0.
 *
 * http://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/
 */

AuditService.$inject = ['$http', 'config'];

function AuditService($http, config) {
  return {

    getPagedAuditEvents(page, sort, size, username, fromDate, toDate) {
      const params = {
        page: page,
        size: size,
        sort: sort,
        username: username,
        fromDate: fromDate ? (moment(fromDate).format('YYYY-MM-DD')) : null,
        toDate: toDate ? (moment(toDate).format('YYYY-MM-DD')) : null,
      };

      return $http({
        url: config.basePath + '/allAudit',
        method: 'GET',
        params: params
      })
    },

    searchOptions() {
      return [{
        name: 'username',
        description: 'Username or email',
        hint: 'Enter username or email',
        maxLength: '50'
      }];
    }

  };
}

angular.module('GLA')
  .service('AuditService', AuditService);

File: audit-service-client/src/main/java/uk/gov/london/ops/audit/AuditService.java
/**
 * Copyright (c) Greater London Authority, 2016.
 *
 * This source code is licensed under the Open Government Licence 3.0.
 *
 * http://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/
 */
package uk.gov.london.ops.audit;

import uk.gov.london.ops.framework.EntityType;

import java.math.BigDecimal;
import java.util.List;

public interface AuditService {

    void auditCurrentUserActivity(String summary);

    void auditCurrentUserActivity(EntityType entityType, Integer entityId, ActivityType type);

    void auditCurrentUserActivity(String summary,  ActivityType type, String userAffected, Integer organisationId, BigDecimal amount);

    void auditActivityForUser(String userName, String summary);

    void auditActivityForUserNewTransaction(String userName, String summary);

    List<AuditableActivity> findAllBySummaryContainingIgnoreCase(String summary);

}
